File: Tyche.Android/session/src/main/java/com/felipearpa/tyche/session/authentication/domain/AuthenticationExternalDataSource.kt
package com.felipearpa.tyche.session.authentication.domain

typealias ExternalAccountId = String

interface AuthenticationExternalDataSource {
    suspend fun sendSignInLinkToEmail(email: String)
    suspend fun signInWithEmailLink(email: String, emailLink: String): ExternalAccountId
    suspend fun isSignInWithEmailLink(emailLink: String): Boolean
    suspend fun signOut()
}

File: Tyche.iOS/Session/Sources/Session/Authentication/Domain/AuthenticationExternalDataSource.swift
typealias ExternalAccountId = String

protocol AuthenticationExternalDataSource {
    func sendSignInLinkToEmail(email: String) async throws
    func signInWithEmailLink(email: String, emailLink: String) async throws -> ExternalAccountId
    func signInWithEmailAndPassword(email: String, password: String) async throws -> ExternalAccountId
    func isSignInWithEmailLink(emailLink: String) async -> Bool
    func signOut() async throws
}
